api/caching/tasks.py

```diff
@@ -131,11 +132,27 @@ def update_storage_usage_cache(target_id, target_guid, per_page=500000):
             offset += count
 
     key = cache_settings.STORAGE_USAGE_KEY.format(target_id=target_guid)
-    storage_usage_cache.set(key, storage_usage_total, cache_settings.FIVE_MIN_TIMEOUT)
+    storage_usage_cache.set(key, storage_usage_total, cache_settings.ONE_DAY_TIMEOUT)
 
 
 def update_storage_usage(target):
     Preprint = apps.get_model('osf.preprint')
 
     if settings.ENABLE_STORAGE_USAGE_CACHE and not isinstance(target, Preprint) and not target.is_quickfiles:
         enqueue_postcommit_task(update_storage_usage_cache, (target.id, target._id,), {}, celery=True)
+
+def update_storage_usage_with_size(target_node, target_file_id, action, size=0,):
+    BaseFileNode = apps.get_model('osf.basefilenode')
+
+    current_usage = target_node.storage_usage or 0
+    target_file = BaseFileNode.load(target_file_id)
+
+    if action in ['create', 'update']:
```

Comment:
What happens in the case of a move/rename? Is it a create then delete? Or would that potentially cause it to count a file twice?

Reply:
The move is handled separately (I'm dealing with that right now). The rename I'm not sure of yet. I'll add it to my list of cases to check.